tests/ragger/keychain.py

```diff
@@ -0,0 +1,23 @@
+import os
+import hashlib
+from ecdsa.util import sigencode_der
+from ecdsa import SigningKey
+from enum import Enum, auto
+
+class Key(Enum):
+    CAL = auto()
+    DOMAIN_NAME = auto()
+
+_keys: dict[Key, SigningKey] = dict()
+
+def _init_key(key: Key):
+    global _keys
+    with open("%s/keychain/%s.pem" % (os.path.dirname(__file__), key.name.lower())) as pem_file:
+        _keys[key] = SigningKey.from_pem(pem_file.read(), hashlib.sha256)
+    assert (key in _keys) and (_keys[key] != None)
+
+def sign_data(key: Key, data: bytes) -> bytes:
+    global _keys
+    if key not in _keys:
+        _init_key(key)
+    return _keys[key].sign_deterministic(data, sigencode=sigencode_der)
```

Comment:
This code is hard to understand, add a small comment to describe what we're doing

Reply:
Added some comments, let me know if they are enough

Reply:
Yes with one small suggestion